docs/ai/how-to/app-service-aoai-auth.md

```diff
@@ -0,0 +1,160 @@
+
+---
+
+# Authenticate and authorize App Service to Azure OpenAI using Microsoft Entra and the Semantic Kernel SDK
+
+This article demonstrates how to use [Microsoft Entra managed identities](/azure/app-service/overview-managed-identity) to authenticate and authorize an App Service application to an Azure OpenAI resource.
+
+By using a managed identity from Microsoft Entra, your App Service application can easily access protected Azure OpenAI resources without having to manually provision or rotate any secrets.
+
+The [Semantic Kernel SDK](/semantic-kernel/overview) enables you to easily implement Microsoft Entra authentication in your .NET application.
+
+## Prerequisites
+
+* An Azure account with an active subscription. [Create an account for free](https://azure.microsoft.com/free/?WT.mc_id=A261C142F).
+* [.NET SDK](https://dotnet.microsoft.com/download/visual-studio-sdks)
+* [`Microsoft.SemanticKernel` NuGet package](https://www.nuget.org/packages/Microsoft.SemanticKernel)
+* [`Azure.Identity` NuGet package](https://www.nuget.org/packages/Azure.Identity)
+* [Create and deploy an Azure OpenAI Service resource](/azure/ai-services/openai/how-to/create-resource)
+* [Create and deploy a .NET application to App Service](/azure/app-service/quickstart-dotnetcore)
+
+## Add a managed identity to App Service
+
+Your application can be granted two types of identities:
+
+* A **system-assigned identity** is tied to your application and is deleted if your app is deleted. An app can only have one system-assigned identity.
+* A **user-assigned identity** is a standalone Azure resource that can be assigned to your app. An app can have multiple user-assigned identities.
+
+### Add a system-assigned identity
+
+:::zone target="docs" pivot="azure-portal"
+
+1. Navigate to your app's page on the [Azure Portal](https://aka.ms/azureportal), scroll down to the **Settings** group.
+1. Select **Identity**.
+1. Within the **System assigned** tab, toggle *Status* to **On**. Click **Save**.
+
+:::zone-end
+
+:::zone target="docs" pivot="azure-cli"
+
+Run the `az webapp identity assign` command to create a system-assigned identity:
+
+```azurecli
+az webapp identity assign --name <app-name> --resource-group <group-name>
+```
+
+:::zone-end
+
+### Add a user-assigned identity
+
+To create an app with a user-assigned identity, first create the identity and then add its resource identifier to your app config.
+
+:::zone target="docs" pivot="azure-portal"
+
+1. Create a user-assigned managed identity resource according to these [instructions](/azure/active-directory/managed-identities-azure-resources/how-to-manage-ua-identity-portal#create-a-user-assigned-managed-identity).
+1. In the left navigation for your app's page, scroll down to the **Settings** group.
+1. Select **Identity**.
+1. Select **User assigned > Add**.
+1. Search for the identity you created earlier, select it, and then select **Add**.
+
+    > [!CAUTION]
+    > Once you select **Add**, the app restarts.
```

Comment:
The impact of the app restarting would depend on the app itself, I'll change this to important because it's up to the dev to decide if that's an issue or not.